### PR TITLE
Fix KVP testcase for FreeBSD

### PR DESCRIPTION
**Pull request by dsrivastavv**

### CODE REVIEW

File: microsoft/testsuites/core/kvp.py [hunk 1]
```diff
@@ -65,31 +74,36 @@ def verify_kvp(self, node: Node, log: Logger) -> None:
...
-        )
-        hv_kvp = get_matched_str(result.stdout, self._hv_kvp_pattern)
-        assert_that(
-            hv_kvp,
-            f"Cannot find hv_kvp in '/proc/{kvp_daemon_pid}/fd'. "
-            "the kernel may not support hv_kvp.",
-        ).is_equal_to("hv_kvp")
+        if not isinstance(node.os, BSD):
+            result = node.execute(
+                f"ls -al /proc/{kvp_daemon_pid}/fd",
```
Comment (dsrivastavv) [inline microsoft/testsuites/core/kvp.py:88]: @whu2014 Can you please clarify if this check is valid for BSD?

**Comment by dsrivastavv** [comment]
@squirrelsc Got a confirmation from Wei that this PR is good to go

File: lisa/tools/pgrep.py [hunk 2]
```diff
@@ -42,3 +46,28 @@ def get_processes(self, process_identifier: str) -> List[ProcessInfo]:
...
+    @property
+    def command(self) -> str:
+        return "ps"
+
+    def get_processes(self, process_identifier: str) -> List[ProcessInfo]:
+        running_process: List[ProcessInfo] = []
+        output = self.run("-axceo user,pid,command", sudo=True, force_run=True).stdout
+        found_processes = find_patterns_in_lines(output, [self._process_map_regex])
+        running_process.extend(
+            ProcessInfo(name=item[2], pid=item[1]) for item in found_processes[0]
```
Comment (squirrelsc) [inline lisa/tools/pgrep.py]: Try to use names instead of like `item[2]`

Comment (dsrivastavv) [reply lisa/tools/pgrep.py]: I am guessing you mean something like `ProcessInfo(name=item.name, pid=item.pid)` but `item` here is a tuple. Can you please clarify?

Comment (squirrelsc) [reply lisa/tools/pgrep.py]: Use `find_groups_in_lines`, so it supports named items.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**